reasoner/planner/RecursivePlanner.java

```diff
@@ -373,10 +374,6 @@ private void porDFS(Stack<Resolvable<?>> currentPath, Set<Variable> currentBound
                     .filter(r -> ReasonerPlanner.dependenciesSatisfied(r, currentBoundVars, dependencies) && !sleepSet.contains(r))
                     .toList();
 
-            if (enabled.isEmpty()) {
-                enabled = iterate(remaining).filter(r -> !r.isNegated() && !sleepSet.contains(r)).toList();
```

Comment:
When our dependencies were too strict, we'd relax them to be able to generate orderings.
Now that they're strict enough, We should prevent that.